src/cbapi/psc/threathunter/models.py

```diff
@@ -175,3 +177,694 @@ def children(self):
+        if not self._report_id:
+            raise InvalidObjectError("ignoring only applies to watchlist IOCs")
+
+        url = "/threathunter/watchlistmgr/v1/report/{}/ioc/{}/ignore".format(self._report_id, self.id)
+        self._cb.delete_object(url)
+
+
+class Watchlist(FeedModel):
+    """Represents a ThreatHunter watchlist.
+    """
+    # NOTE(ww): Not documented.
+    urlobject = "/threathunter/watchlistmgr/v2/watchlist"
+    urlobject_single = "/threathunter/watchlistmgr/v2/watchlist/{}"
+    swagger_meta_file = "psc/threathunter/models/watchlist.yaml"
+
+    @classmethod
+    def _query_implementation(cls, cb):
+        return WatchlistQuery(cls, cb)
+
+    def __init__(self, cb, model_unique_id=None, initial_data=None):
+        item = {}
+
+        if initial_data:
+            item = initial_data
+        elif model_unique_id:
+            # TODO(ww): It's probably bad practice to make a request here.
+            # Maybe abstract this into a separate method?
+            item = cb.get_object(self.urlobject_single.format(model_unique_id))
+
+        feed_id = item.get("id")
+
+        super(Watchlist, self).__init__(cb, model_unique_id=feed_id, initial_data=item,
+                                        force_init=False, full_doc=True)
+
+    def save(self):
+        """Saves this watchlist on the ThreatHunter server.
+
+        :return: The saved watchlist
+        :rtype: :py:class:`Watchlist`
+        :raise InvalidObjectError: if :py:meth:`validate` fails
+        """
+        self.validate()
+
+        new_info = self._cb.post_object("/threathunter/watchlistmgr/v2/watchlist", self._info).json()
+        self._info.update(new_info)
+        return self
+
+    def validate(self):
+        """Validates this watchlist's state.
+
+        :raise InvalidObjectError: if the watchlist's state is invalid
+        """
+        super(Watchlist, self).validate()
+
+    def update(self, **kwargs):
+        """Updates this watchlist with the given arguments.
+
+        >>> watchlist.update(name="New Name")
+
+        :param kwargs: The fields to update
+        :type kwargs: dict(str, str)
+        :raise InvalidObjectError: if `id` is missing or :py:meth:`validate` fails
+        :raise ApiError: if `report_ids` is given *and* is empty
+        """
+        if not self.id:
+            raise InvalidObjectError("missing Watchlist ID")
+
+        # NOTE(ww): Special case, according to the docs.
+        if "report_ids" in kwargs and not kwargs["report_ids"]:
+            raise ApiError("can't update a watchlist to have an empty report list")
+
+        for key, value in kwargs.items():
+            if key in self._info:
+                self._info[key] = value
+
+        self.validate()
+
+        new_info = self._cb.put_object("/threathunter/watchlistmgr/v2/watchlist/{}".format(self.id), self._info).json()
+        self._info.update(new_info)
+
+    @property
+    def classifier_(self):
```

Comment:
is this underscored because there is already an `_info` field called classifier?

Reply:
Yup, exactly. It's the same issue as I marked with the `NOTE` above.